src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -1,5 +1,109 @@
 package frc.robot.subsystems;
 
-public class Swerve {
+import static edu.wpi.first.units.Units.Degrees;
+import static edu.wpi.first.units.Units.Inches;
+import static edu.wpi.first.units.Units.Meters;
 
+import java.io.File;
+import java.io.IOException;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.SwerveK;
+import swervelib.SwerveDrive;
+import swervelib.math.SwerveMath;
+import swervelib.parser.SwerveParser;
+import swervelib.telemetry.SwerveDriveTelemetry;
+import swervelib.telemetry.SwerveDriveTelemetry.TelemetryVerbosity;
+
+public class Swerve extends SubsystemBase {
+
+    SwerveDrive swerveDrive;
+    double maximumSpeed = Units.feetToMeters(4.5);
+
+    /**
+     * 
+     * @param directory Directory of swerve drive config files.
+     */
+    public Swerve(File directory) throws IOException {
+        double angleConversionFactor = SwerveMath.calculateDegreesPerSteeringRotation(SwerveK.steerGearRatio);
+        double driveConversionFactor = SwerveMath.calculateMetersPerRotation(SwerveK.driveGearRatio, SwerveK.wheelDiameter.in(Inches));
+
+        SwerveDriveTelemetry.verbosity = TelemetryVerbosity.HIGH;
+
+        swerveDrive = new SwerveParser(directory).createSwerveDrive(maximumSpeed, angleConversionFactor, driveConversionFactor);
+    }
+
+    public void setupPathPlanner() {
+        AutoBuilder.configureHolonomic(
+            this::getPose, 
+            this::resetOdometry, 
+            this::getRobotVelocity, 
+            this::setChassisSpeeds, 
+            new HolonomicPathFollowerConfig(SwerveK.translationConstants, 
+                                            SwerveK.rotationConstants, 
+                                            SwerveK.maxModuleSpeed, 
+                                            SwerveK.driveBaseRadius.in(Meters), 
+                                            null), 
+            null,
```

Comment:
Looks like this parameter is for deciding when to mirror a path for being on red alliance, example code is [here](https://pathplanner.dev/pplib-getting-started.html#holonomic-swerve)